test/intrinsics.jl

```diff
@@ -215,18 +281,60 @@ end
     @test_intrinsic Core.Intrinsics.fptoui UInt Float32(3.3) UInt(3)
 end
 
+function f16(sign, exp, sig)
+    x = (sign&1)<<15 | (exp&((1<<5)-1))<<10 | sig&((1<<10)-1)
+    return reinterpret(Float16, UInt16(x))
+end
+function f32(sign, exp, sig)
+    x = (sign&1)<<31 | (exp&((1<<8)-1))<<23 | sig&((1<<23)-1)
+    return reinterpret(Float32, UInt32(x))
+end
+function f64(sign, exp, sig)
+    x = (sign&1)<<31 | (exp&((1<<11)-1))<<52 | sig&((1<<52)-1)
+    return reinterpret(Float64, UInt64(x))
+end
```

Comment:
This is wrong on 32 bit systems (you want `x = Int64(sign&1)<<31 | Int64(exp&((1<<11)-1))<<52 | sig&((Int64(1)<<52)-1)`